zerver/lib/test_classes.py

```diff
@@ -1313,6 +1314,26 @@ def check_user_subscribed_only_to_streams(self, user_name: str, streams: List[St
         for x, y in zip(subscribed_streams, streams):
             self.assertEqual(x["name"], y.name)
 
+    def mark_topic_resolved(
+        self,
+        user: UserProfile,
```

Comment:
I might call this variable `acting_user`, since that's a term we often use for that concept; but there's not really another role the user might have in this function.

But having thought about that for a moment, I think name for the function might not be right, since it doesn't accept a "topic" as a parameter. Maybe `resolve_topic_containing_message(target_message_id, acting_user)`?

If we were going to use this in 5 places one could think about things like making `propagate_mode` a parameter, but I think given we don't have a lot of uses, we can leave that to someone who is adding them.

Reply:
I like that - kind of a part two of the naming mini-crisis I called out in my self-review there. *Something* felt off about how it was written but I had the "wrote the thing so it makes sense here?" blinders on. Thanks!

`propagate_mode` feels like a great thing to have in an Enum class somewhere shared. That _also_ feels out of scope for this PR (albeit cheap to implement), so for now I'll punt to a `#backend` chat.